src/pages/index.js

```diff
@@ -54,55 +73,177 @@ const profileEditModal = new ModalWithForm(
 const previewImageModal = new ModalWithImage("#preview-image-modal");
 previewImageModal.setEventListeners();
 
+// Handle profile avatar update function
+function handleProfileAvatarUpdate(inputValues) {
+  console.log("inputValues:", inputValues);
+  console.log("inputValues.avatar:", inputValues.avatar);
+
+  profileAvatarEditModal.setModalLoad(true);
+  api
+    .updateUserAvatar(inputValues.avatar)
+    .then((data) => {
+      // Update the avatar image
+      userInfo.setAvatar(data.avatar);
+      avatarFormValidator.disableButton();
+      profileAvatarEditModal.close();
+      profileAvatarForm.reset();
+    })
+    .catch((err) => {
+      console.error("Error updating avatar:", err);
+
+      const errorMessage = document.querySelector(
+        "#profile-avatar-input-error"
+      );
+      errorMessage.textContent = "Failed to update avatar. Please try again.";
+    })
+    .finally(() => {
+      profileAvatarEditModal.setModalLoad(false);
+    });
+}
+
 /*    Functions for opening/closing modals    */
 function handleProfileEditSubmit(data) {
-  userInfo.setUserInfo({ name: data.name, description: data.description });
-  profileEditModal.close();
+  profileEditModal.setModalLoad(true);
+  api
+    .updateUserInfo(data.name, data.description)
+    .then(() => {
+      // Update the UI with the new user data
+      userInfo.setUserInfo(data);
+      editFormValidator.disableButton();
+      profileEditModal.close();
+    })
+    .catch((err) => {
+      console.error("Error updating user info:", err);
+    })
+    .finally(() => {
+      profileEditModal.setModalLoad(false);
+    });
 }
 
 function handleAddCardCreate(data) {
   const name = data.title;
   const link = data.link;
+  addCardModal.setModalLoad(true);
+  api
+    .addCard(name, link)
+    .then((cardData) => {
+      renderCard(cardData);
+      addFormValidator.disableButton();
+      addCardForm.reset();
+      addCardModal.close();
+    })
+    .catch((err) => {
+      console.error("Error adding card:", err);
+    })
+    .finally(() => {
+      addCardModal.setModalLoad(false);
+    });
+}
 
-  console.log(`Creating card with name: ${name} and link: ${link}`);
+const deleteConfirmationModal = new ModalWithConfirmation({
+  modalSelector: "#delete-confirmation-modal",
+});
 
-  renderCard({ name, link });
-  addCardModal.close();
-  addCardForm.reset();
-  addFormValidator.disableButton();
+deleteConfirmationModal.setEventListeners();
+
+function handleDeleteCard(card) {
+  deleteConfirmationModal.open();
+  deleteConfirmationModal.setSubmitFunction(() => {
+    api
+      .deleteCard(card._id)
+      .then(() => {
+        card.handleRemoveCard();
+        deleteConfirmationModal.close();
+      })
+      .catch((err) => {
+        console.error("Error deleting card:", err);
+      });
+  });
+}
+
+function handleImageClick(name, link) {
+  previewImageModal.open({ name, link });
+}
+
+function handleLikeCard(card) {
+  if (card._isLiked) {
```

Comment:
`_isLiked`

Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.